### ISSUE TITLE
v1.4.0-rc5 cannot push docker images

### ISSUE THREAD

**Issue by saschagrunert** [containous/traefik#2246]
Since rc5 of traefik a push to a docker registry behind the proxy will fail:

```console
> docker push my.registry.com/test:latest
The push refers to a repository [my.registry.com/test:latest]
d1b1751158bf: Retrying in 1 second
16d7333caf66: Retrying in 1 second
e11f893dcbca: Retrying in 2 seconds
9d352487e7d2: Retrying in 1 second
56a71a6a57ef: Pushing [=>                                                 ]     10MB/271.2MB
0885393eaa8d: Waiting
26d0f604dc1b: Waiting
```

After some retries the push aborts. Traefik reports the following errors:
```shell
time="2017-10-11T11:53:15Z" level=warning msg="Error forwarding to https://10.2.13.13:443, err: stream error: stream ID 87; PROTOCOL_ERROR" 
time="2017-10-11T11:53:15Z" level=warning msg="Error forwarding to https://10.2.13.13:443, err: stream error: stream ID 89; PROTOCOL_ERROR" 
time="2017-10-11T11:53:15Z" level=warning msg="Error forwarding to https://10.2.13.13:443, err: stream error: stream ID 91; PROTOCOL_ERROR" 
time="2017-10-11T11:53:17Z" level=warning msg="Error forwarding to https://10.2.13.13:443, err: stream error: stream ID 121; PROTOCOL_ERROR" 
```
I am using registry:2.6.2 and traefik with acme.

I've tested the same with 1.3.8 which works as expected. My traefik config:

```toml
logLevel = "INFO"
InsecureSkipVerify = true
defaultEntryPoints = ["http", "https"]
[entryPoints]
  [entryPoints.http]
  address = ":80"
    [entryPoints.http.redirect]
    entryPoint = "https"
  [entryPoints.https]
  address = ":443"
  compress = true
    [entryPoints.https.tls]
      [[entryPoints.https.tls.certificates]]
      CertFile = "/ssl/tls.crt"
      KeyFile = "/ssl/tls.key"
[kubernetes]
[web]
address = ":8181"
CertFile = "/ssl/tls.crt"
KeyFile = "/ssl/tls.key"
  [web.auth.basic]
    users = ["<removed>:<removed>"]
  [web.statistics]
    RecentErrors = 50
  [web.metrics.prometheus]
    Buckets=[0.1,0.3,1.2,5.0]
[acme]
email = "<removed>"
storage = "/acme/acme.json"
entryPoint = "https"
onHostRule = true
```

**Comment by nmengin** [comment]
Hello @saschagrunert .

Thanks for your feedback!
In the way to analyze the probelm, I would like to reproduce your environment.

That's why it could be interesting to know the Træfik configuration for your registry pod.
Can you share it?

**Comment by saschagrunert** [comment]
Hey @nmengin, sure. Please be aware that you need to generate the TLS certs (or disable tls for testing) and the [htpasswd for access](https://docs.docker.com/registry/deploying/#restricting-access):

```yml
---
apiVersion: v1
kind: Secret
type: Opaque
metadata:
  name: registry-login
  namespace: kube-system
data:
  htpasswd: <base64 encoded registry htpasswd content>
---
apiVersion: v1
kind: Secret
type: kubernetes.io/tls
metadata:
  name: registry-server
  namespace: kube-system
data:
  tls.crt: <base64 encoded certificate>
  tls.key: <base64 encoded key>
---
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  labels:
    k8s-app: registry
  name: registry
  namespace: kube-system
spec:
  replicas: 1
  selector:
    matchLabels:
      k8s-app: registry
  template:
    metadata:
      labels:
        k8s-app: registry
    spec:
      containers:
        - env:
            - name: REGISTRY_HTTP_ADDR
              value: 0.0.0.0:443
            - name: REGISTRY_STORAGE_FILESYSTEM_ROOTDIRECTORY
              value: /var/lib/registry
            - name: REGISTRY_HTTP_TLS_CERTIFICATE
              value: /certs/tls.crt
            - name: REGISTRY_HTTP_TLS_KEY
              value: /certs/tls.key
            - name: REGISTRY_AUTH
              value: htpasswd
            - name: REGISTRY_AUTH_HTPASSWD_REALM
              value: registry realm
            - name: REGISTRY_AUTH_HTPASSWD_PATH
              value: /auth/htpasswd
          image: registry:2.6.2
          name: registry
          resources:
            limits:
              cpu: 100m
              memory: 100Mi
            requests:
              cpu: 100m
              memory: 100Mi
          volumeMounts:
            - mountPath: /var/lib/registry
              name: image-store
            - mountPath: /certs
              name: registry-server
            - mountPath: /auth
              name: registry-login
      nodeSelector:
        registry: "true"
      volumes:
        - name: image-store
          hostPath:
            path: /srv/registry
        - name: registry-server
          secret:
            defaultMode: 420
            secretName: registry-server
        - name: registry-login
          secret:
            defaultMode: 420
            secretName: registry-login
---
apiVersion: v1
kind: Service
metadata:
  name: registry
  namespace: kube-system
spec:
  selector:
    k8s-app: registry
  ports:
    - port: 443
      protocol: TCP
  type: ClusterIP
---
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  annotations:
    kubernetes.io/ingress.class: traefik
  name: registry
  namespace: kube-system
spec:
  rules:
    - host: registry.domain.com
      http:
        paths:
          - backend:
              serviceName: registry
              servicePort: 443
```

**Comment by nmengin** [comment]
Many thanks @saschagrunert.

Unfortunately, your issue is quiet difficult to reproduce...

It can help if we know from when there is this problem.
Do you think you can check with previous RC versions? As 1.4.0-rc4 for example (image : traefik:1.4.0-rc4)

**Comment by saschagrunert** [comment]
Hey, yep I‘ve used 1.4.0-rc3 1.3.X for quite a while without any problems. I’m glad that you found the issue. 🙂

**Comment by traefiker** [comment]
Closed by #2258.
